plugins/modules/docker_swarm.py

```diff
@@ -507,6 +521,7 @@ def init_swarm(self):
                 'advertise_addr': self.parameters.advertise_addr,
                 'listen_addr': self.parameters.listen_addr,
                 'data_path_addr': self.parameters.data_path_addr,
+                'data_path_port': self.parameters.data_path_port,
```

Comment:
You have to do the same `if` dance as for `default_addr_pool` and `subnet_size` to add this one, since it is not supported by older Docker SDK for Python versions.